luarules/gadgets/unit_custom_weapons_cluster.lua

```diff
@@ -159,140 +148,212 @@ for wdid, data in pairs(dataTable) do
             dataTable[wdid].def .. ' is not spawnable (' .. (cmdef.type or 'nil!') .. ')')
         dataTable[wdid] = nil
     end
+
+    -- Remove invalid spawn counts:
+    if data.number == nil or data.number <= 1 then
+        Spring.Echo('[clusermun] [warn] Removing low-count cluster weapon: ' .. wdid)
+        dataTable[wdid] = nil
+    end
 end
 
--- Information on how sturdy units are, basically
+-- Information on units
+
+local unitBulk = {} -- How sturdy the unit is. Projectiles scatter less with lower bulk values.
 
-local unitBulk = {}
 for udid, udef in pairs(UnitDefs) do
-    unitBulk[udid] = min(
-        (   sqrt(udef.health) +
-            sqrt(udef.metalCost) * udef.xsize * udef.zsize * sqrt(udef.radius)
-        ) / minBulkReflect,
-        1.0
-    )
+    -- Set the unit bulk values.
+    -- todo: We don't even _detect_ walls. "Objectified" units aren't returned by GetUnitsInSphere?
+    -- todo: Seems likely that the same goes for wall-like units, then.
+    if udef.armorType == Game.armorTypes.wall or udef.armorType == Game.armorTypes.indestructible then
+        unitBulk[udid] = 0.9
+    elseif udef.name == 'armclaw' or udef.name == 'cormaw' or udef.name == 'legjaw' then
```

Comment:
Being a dragon turret could be a customparam, perhaps bulk as well

Reply:
Picked up dragon turrets by customparam. Left bulk in its local table for now.